### PR TITLE
Fix sourceSets for IntelliJ.

### PR DESCRIPTION
**Pull request by mezz**
This PR adds/removes the following features:
* Creates a plain mod jar, an API jar, ~~and an "all" jar which contains both.~~

It changes the following features:
* API jar now includes compiled code and sources for distribution.

It fixes the following bugs:
* Packaging the API jar this way works around a ForgeGradle issue https://github.com/MinecraftForge/ForgeGradle/issues/369
* IntelliJ sourceSets were not properly separated, now the api folder is [api] and main is [main]
Before pic:
idea64_2017-03-15_22-23-48

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -154,21 +155,34 @@ processResources {
...
+    // https://github.com/MinecraftForge/ForgeGradle/issues/369
+    // Gradle should be able to pull them from the -sources jar.
+    from sourceSets.api.allJava
+}
+
+task allJar(type: Jar) {
+    classifier = 'all'
+    from sourceSets.main.output
+    from (sourceSets.api.output) {
+        exclude("logoblank.png", "mcmod.info")
```
Comment (elifoster) [inline build.gradle]: Why are these files excluded?

Comment (mezz) [reply build.gradle]: The resources from the `api` sourceSet do not belong in this jar, they would be duplicates of the ones in `main`.

Comment (mezz) [reply build.gradle]: Maybe I should use `sourceSets.api.allJava` instead?

Comment (mezz) [reply build.gradle]: It's possible that this jar just shouldn't exist, since it would have 2 `@Mod`s in it.

Comment (elifoster) [reply build.gradle]: ah right. I think either'd do tbh. Although, I planned on adding actual resources to the API resource set, like crafting gui images for the book. Is that actually possible, and if so, would they be in their own `esteemedinnovationapi` dir?

Comment (elifoster) [reply build.gradle]: I am pretty sure you can have multiple `@Mod`s in a single jar no problem. I can't remember if I tested a fully compiled non-dev environment instance though.

Comment (mezz) [reply build.gradle]: I will remove this jar for now, there is still a bit to figure out for it.

**Comment by elifoster** [comment]
@mezz Can you rebase this?

**Pull request closed without merging**